core/clipboard/block_paster.ts

```diff
@@ -55,7 +56,11 @@ export class BlockPaster implements IPaster<BlockCopyData, BlockSvg> {
     if (eventUtils.isEnabled() && !block.isShadow()) {
       eventUtils.fire(new (eventUtils.get(EventType.BLOCK_CREATE))(block));
     }
-    common.setSelected(block);
+
+    // Sometimes there's a delay before the block is fully created and ready for
+    // focusing, so wait slightly before focusing the newly pasted block.
+    const nodeToFocus: IFocusableNode = block;
+    setTimeout(() => getFocusManager().focusNode(nodeToFocus), 0);
```

Comment:
This delay may be related to the `bumpNeighbours` behaviour.

Reply:
Hmm that seems really plausible, though I can't find a call to bump in this call path but I'm sure I'm missing large portions of what's happening here.

This does make me think that it might be nice to have a way to attach logic to run with the next rendering pass so that we can predictably run things that depend on rendering (vs. using a timeout and hoping it's correct).

Reply:
```
/**
 * @returns A promise that resolves after the currently queued renders have
 *     been completed.
 */
export function finishQueuedRenders(): Promise<void> {
  // If there are no queued renders, return a resolved promise so `then`
  // callbacks trigger immediately.
  return afterRendersPromise ? afterRendersPromise : Promise.resolve();
}
```

You can find examples of this pattern in the codebase. I think it will do what you want.

Reply:
I evidently didn't look hard enough for this exact mechanism. Nice suggestion--I updated the code and tested pasting again. Selection seems to correctly select the most recently pasted block as intended.